framework/wcf/Extensibility/Transport/AzureQueueStorageClient/AzureQueueStorageClient/AzureQueueStorageOutputChannel.cs

```diff
@@ -0,0 +1,263 @@
+        {
+        }
+
+        protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            return new CompletedAsyncResult(callback, state);
+        }
+
+        protected override void OnEndOpen(IAsyncResult result)
+        {
+            CompletedAsyncResult.End(result);
+        }
+
+
+        #region Socket_Shutdown
+        /// <summary>
+        /// Shutdown ungracefully
+        /// </summary>
+        protected override void OnAbort()
+        {
+
+        }
+
+        /// <summary>
+        /// Shutdown gracefully
+        /// </summary>
+        protected override void OnClose(TimeSpan timeout)
+        {
+
+        }
+
+        protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            this.OnClose(timeout);
+            return new CompletedAsyncResult(callback, state);
+        }
+
+        protected override void OnEndClose(IAsyncResult result)
+        {
+            CompletedAsyncResult.End(result);
+        }
+        #endregion
+
+        #region Send_Synchronous
+        /// <summary>
+        /// Address the Message and serialize it into a byte array.
+        /// </summary>
+        ArraySegment<byte> EncodeMessage(Message message)
+        {
+            try
+            {
+                this.remoteAddress.ApplyTo(message);
+                return encoder.WriteMessage(message, int.MaxValue, parent.BufferManager);
+            }
+            finally
+            {
+                // We have consumed the message by serializing it, so clean up
+                message.Close();
+            }
+        }
+
+        public void Send(Message message)
+        {
+            try
+            {
+                queueClient.SendMessage(message.ToString());
+            }
+            catch
+            {
+
+            }
+            finally
+            {
+               
+            }
+        }
+
+        public void Send(Message message, TimeSpan timeout)
+        {
+            // UDP does not block so we do not need timeouts.
+            this.Send(message);
+        }
+        #endregion
+
+        #region Send_Asynchronous
+        public IAsyncResult BeginSend(Message message, AsyncCallback callback, object state)
+        {
+            //base.ThrowIfDisposedOrNotOpen();
+            return new SendAsyncResult(this, message, callback, state);
+        }
+
+        public IAsyncResult BeginSend(Message message, TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            // UDP does not block so we do not need timeouts.
+            return this.BeginSend(message, callback, state);
+        }
+
+        public void EndSend(IAsyncResult result)
+        {
+            SendAsyncResult.End(result);
+        }
+
+        /// <summary>
+        /// Implementation of async send for Udp. 
+        /// </summary>
+        class SendAsyncResult : AsyncResult
+        {
```

Comment:
This class goes away when using an async method and using TaskHelpers.ToApm and ToApmEnd.

Reply:
removed.